runtime.md

```diff
@@ -1,5 +1,47 @@
 # Runtime and Lifecycle
 
+## State
+
+The runtime state for a container is persisted on disk so that external tools can consume and act on this information.
+The runtime state is stored in a JSON encoded file.
+It is recommended that this file is stored in a temporary filesystem so that it can be removed on a system reboot.
+On Linux based systems the state information should be stored in `/run/oci/containers`.
+The directory structure for a container is `/run/oci/containers/<containerID>/state.json`.
+By providing a default location that container state is stored external applications can find all containers running on a system.
+
+* **version** (string) Version of the OCI specification used when creating the container.
+* **id** (string) ID is the container's ID.
+* **pid** (int) Pid is the ID of the main process within the container.
+* **root** (string) Root is the path to the container's bundle directory.
```

Comment:
On Sat, Aug 29, 2015 at 09:36:51AM -0700, Vincent Batts wrote:
> Deriving the `root` of a container from `/proc/pid/mountinfo` is not
> guaranteed. So this field _is_ needed.

You mentioned root needing to be mounted for this [1], but I'm not
clear on what that means.  In what situation would the container's
root not be mounted?  For example, here's a container with a mount
namespace:

  $ cat /proc/6468/mountinfo 
  67 41 0:18 /home/…/oci-gentoo-minimal/rootfs / ro,noatime - btrfs /dev/sda4 rw,ssd,space_cache
  …

showing that the container's root is at the host's
/home/…/oci-gentoo-minimal/rootfs.

And here's a container without a mount namespace:

  $ cat /proc/6304/mountinfo 
  18 0 0:18 / / rw,noatime - btrfs /dev/sda4 rw,ssd,space_cache
  …

showing that the container's root is the same as my host's root.
Although without a mount namespace, I don't know what the
config.json's ‘root’ setting does (this is similar to, but not the
same as my not thinking that we should require a rootfs directory in
the bundle [2]).

[1]: https://github.com/opencontainers/specs/pull/87/files#r38261424
[2]: https://groups.google.com/a/opencontainers.org/forum/#!topic/dev/6ZKMNWujDhU